x/gov/handler.go

```diff
@@ -179,7 +196,17 @@ func (h TallyHandler) Deliver(ctx weave.Context, db weave.KVStore, tx weave.Tx)
 	if err := proposal.Tally(); err != nil {
 		return nil, err
 	}
-	if err := h.bucket.Update(db, msg.ProposalID, proposal); err != nil {
+
+	if proposal.Result == Proposal_Accepted {
+		exec, ok := h.typeHandler[proposal.Type]
```

Comment:
Ah, here is where we map the execution behavior on success. Seems abstract enough.